containers/Investment/Investor/index.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
+import { AuthState } from '../../../ducks/auth';
+import { InvestorState, loadInvestor } from '../../../ducks/investments';
+import { connect } from 'react-redux';
+import { Box, Tab, Tabs } from 'grommet';
+import Alert from '../../../components/Alert';
+import { Spinner } from '@centrifuge/axis-spinner';
+import { isValidAddress } from '../../../utils/address';
+import TrancheView from '../Tranche';
+import { TransactionState, resetTransactionState } from '../../../ducks/transactions';
+import { AnalyticsState, loadAnalyticsData } from '../../../ducks/analytics';
+
+interface Props {
+  tinlake: any;
+  auth: AuthState;
+  loadInvestor?: (tinlake: any, address: string) => Promise<void>;
+  investments?: InvestorState;
+  transactions?: TransactionState;
+  resetTransactionState?: () => void;
+  loadAnalyticsData?: (tinlake: any) => Promise<void>;
+  analytics?: AnalyticsState;
+  investorAddress: string;
+}
+
+interface State {
+  errorMsg: string;
+  is: string | null;
+}
+
+class InvestorView extends React.Component<Props, State> {
+
+  showInvestor = async () => {
+    const { investorAddress } = this.props;
+    const { loadInvestor, tinlake } = this.props;
+    resetTransactionState && resetTransactionState();
+
+    this.setState({ is: null, errorMsg: '' });
+    if (!isValidAddress(investorAddress)) {
+      this.setState({ is: 'error', errorMsg: 'Please input a valid Ethereum address.' });
+      return;
+    }
+    loadInvestor && loadInvestor(tinlake, investorAddress);
+  }
+
+  componentWillMount() {
+    const { resetTransactionState, loadAnalyticsData, tinlake } = this.props;
+    resetTransactionState && resetTransactionState();
+    loadAnalyticsData && loadAnalyticsData(tinlake);
+    this.setState({ is: null, errorMsg: '' });
```

Comment:
I think this setState is redundant bc it is called in `this.showInvestor` right after